Makefile

```diff
@@ -0,0 +1,300 @@
+	@echo -e -n "$(YELLOW)"
+	@$(VENV_DIR)/$(1)/bin/python3 -m pip install --upgrade pip setuptools wheel
+	@echo -e "$(RESET_COLOR)"
+	@echo -n "Python setup tools updated..."
+	@echo -e "$(OK_MSG)"
+	@echo -e -n "$(YELLOW)"
+	@test ! -s $(2)pre_requirements.txt ||\
+ 		($(VENV_DIR)/$(1)/bin/pip install --upgrade -r $(2)pre_requirements.txt\
+ 		&& echo -e "$(RESET_COLOR)"\
+ 		&& echo -n "Pre requirements installed..."\
+ 		&& echo -e "$(OK_MSG)")
+	@echo -e -n "$(YELLOW)"
+	@test ! -s $(2)requirements.txt ||\
+		($(VENV_DIR)/$(1)/bin/pip install --upgrade -r $(2)requirements.txt\
+ 		&& echo -e "$(RESET_COLOR)"\
+ 		&& echo -n "Requirements installed..."\
+ 		&& echo -e "$(OK_MSG)")
+	@echo -e -n "$(RESET_COLOR)"
+	@test ! -s $(2)setup.py ||\
+		(echo "Installing the package..."\
+		 && echo -e "$(YELLOW)"\
+		 && $(VENV_DIR)/$(1)/bin/pip install --upgrade -e .$(PIP_ARGS)\
+		 && echo -e "$(RESET_COLOR)"\
+		 && echo -n "Package installation completed..."\
+		 && echo -e "$(OK_MSG)")
+	@echo -e "$(RESET_COLOR)"
+	$(call check_license,$(1))
+endef
+
+
+define install_connectors
+	echo
+	echo "--------------------------------------------------------------------------"
+	echo "Installing $1 connector..."
+	echo "--------------------------------------------------------------------------"
+	if [[ ! -d singer-connectors/$1 ]]; then\
+		echo "ERROR: Directory not exists and does not look like a valid singer connector: singer-connectors: singer-connectors/$1";\
+		exit 1;\
+    fi
+    $(call make_virtualenv,$1,singer-connectors/$1/)
+endef
+
+define print_list_of_connectors
+	echo "   $1"
+endef
+
+help: .pw_logo
+	@echo
+	@echo "  Targets"
+	@echo "  ======="
+	@echo "     pipelinewise                                               Install the main PipelineWise component"
+	@echo "     pipelinewise_no_test_extras                                Install the main Pipelinewise component without test extras"
+	@echo
+	@echo "     all_connectors                                             Install all connectors"
+	@echo "     extra_connectors"                                          Install only extra connectors
+	@echo "     connectors -e pw_connector=connector1,connector2,...       Install specific connector(s)"
+	@echo
+	@echo "     list_components                                            Show a list of installed components"
+	@echo "     list_default_connectors                                    Show a list of available default connectors"
+	@echo "     list_extra_connectors                                      Show a list of available extra connectors"
+	@echo
+	@echo "     clean_all                                                  Clean all installed components"
+	@echo "     clean -e pw_connector=connector1,connector2,...            Clean a specific connector(s)"
+	@echo
+	@echo "   Options"
+	@echo "   ======="
+	@echo "      -e pw_connector=connector1,connector2,...                 Define a list of connectors for installing or cleaning"
```

Comment:
because they are environment variables in fact. I wanted to be sure they are unique and wont make any conflict with other environment variables by mistake.

Reply:
ok